### DISCUSSION TITLE
【命令】将选中文件移入以焦点处文件名称命名的文件夹中

### DISCUSSION THREAD

**Discussion by ttbamov** [Chaoses-Ib/DirectoryOpus#2] [分享]
【功能】
将选中文件移入以焦点处文件名称命名的文件夹中
【说明】
选中文件后，使用命令，会将文件移至一个新的文件夹内
文件夹的名字会有弹窗确认，弹窗内选中的文字为最后鼠标点击选中的文件，可直接进行修改
确认后文件夹会被快速的移至该文件夹内
【命令内容】
```js
function OnClick(clickData)
{
var cmd = clickData.func.command;
cmd.deselect = false;
var focesIten = clickData.func.sourcetab.GetFocusItem;
var name = focesIten.name_stem_m.replace(/\s*(.*/,"")
cmd.AddLine('@set name={dlgstringS|请输入新建文件夹名称|"' + name + '"}');
cmd.AddLine('Copy MOVE HERE CREATEFOLDER="{$name}"');
cmd.Run;
}
```
【压缩包文件】
[将选中文件移入以焦点处文件名称命名的文件夹中.zip](https://github.com/user-attachments/files/16117804/default.zip)
【截图】
![PixPin_2024-07-06_20-05-23](https://github.com/Chaoses-Ib/DirectoryOpus/assets/128291620/61f5447f-6327-4fcb-b8b2-cc32d76d2109)
【演示】
![将选中文件移入以焦点处文件名称命名的文件夹中](https://github.com/Chaoses-Ib/DirectoryOpus/assets/128291620/dfacde19-7069-4861-aa82-787f76c09c22)

by@白天

**Comment by Chaoses-Ib** [comment]
代码粘贴丢了一个 `\`，修复版代码：
```js
function OnClick(clickData)
{
    var cmd = clickData.func.command;
    cmd.deselect = false;
    var focesIten = clickData.func.sourcetab.GetFocusItem;
    var name = focesIten.name_stem_m.replace(/\s*\(.*/,"")
    cmd.AddLine('@set name={dlgstringS|请输入新建文件夹名称|"' + name + '"}');
    cmd.AddLine('Copy MOVE HERE CREATEFOLDER="{$name}"');
    cmd.Run;
}
```

**Reply by param1903** [reply]
> The code was pasted and one was lost `\`, the repaired version of the code:
> 
> ```js
> function OnClick(clickData)
> {
>     var cmd = clickData.func.command;
>     cmd.deselect = false;
>     var focesIten = clickData.func.sourcetab.GetFocusItem;
>     var name = focesIten.name_stem_m.replace(/\s*\(.*/,"")
>     cmd.AddLine('@set name={dlgstringS|请输入新建文件夹名称|"' + name + '"}');
>     cmd.AddLine('Copy MOVE HERE CREATEFOLDER="{$name}"');
>     cmd.Run;
> }
> ```

Thank you for posting the repaired version, it works perfectly, and thank you for always helping
